From 46af2d65490b914c0d4a6a6707ae4c7b57b33eae Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 2 Oct 2020 15:52:42 -0500
Subject: [PATCH 57/62] MLK-21880-1 drm/bridge: sec-dsim: correct DSIM version
 register access

During the DSIM binding stage, only enable cfg clock is not
enough to access the version register, since at this time,
if the DSIM is still in the reset state, the register read
will always return 0. So before the version register read,
the runtime pm should be in resume state.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
---
 drivers/gpu/drm/bridge/sec-dsim.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/bridge/sec-dsim.c b/drivers/gpu/drm/bridge/sec-dsim.c
index c2247afc506d..0dfdbe7eb3d6 100644
--- a/drivers/gpu/drm/bridge/sec-dsim.c
+++ b/drivers/gpu/drm/bridge/sec-dsim.c
@@ -22,6 +22,7 @@
 #include <linux/log2.h>
 #include <linux/module.h>
 #include <linux/of_graph.h>
+#include <linux/pm_runtime.h>
 #include <drm/bridge/sec_mipi_dsim.h>
 #include <drm/drmP.h>
 #include <drm/drm_atomic_helper.h>
@@ -1858,10 +1859,12 @@ int sec_mipi_dsim_bind(struct device *dev, struct device *master, void *data,
 		return ret;
 	}
 
-	clk_prepare_enable(dsim->clk_cfg);
+	dev_set_drvdata(dev, dsim);
+
+	pm_runtime_get_sync(dev);
 	version = dsim_read(dsim, DSIM_VERSION);
 	WARN_ON(version != pdata->version);
-	clk_disable_unprepare(dsim->clk_cfg);
+	pm_runtime_put_sync(dev);
 
 	dev_info(dev, "version number is %#x\n", version);
 
@@ -1915,8 +1918,6 @@ int sec_mipi_dsim_bind(struct device *dev, struct device *master, void *data,
 	bridge->of_node = dev->of_node;
 	bridge->encoder = encoder;
 
-	dev_set_drvdata(dev, dsim);
-
 	/* attach sec dsim bridge and its next bridge if exists */
 	ret = drm_bridge_attach(encoder, bridge, NULL);
 	if (ret) {
-- 
2.17.1

